charts/argo-events/values.yaml

```diff
@@ -1,163 +1,406 @@
-  # -- Resource limits and requests for the event source controller pods
+
+  # -- Assign custom [TopologySpreadConstraints] rules to the events controller
+  ## Ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/
+  ## If labelSelector is left out, it will default to the labelSelector configuration of the deployment
+  topologySpreadConstraints: []
+  # - maxSkew: 1
+  #   topologyKey: topology.kubernetes.io/zone
+  #   whenUnsatisfiable: DoNotSchedule
+
+  # -- Priority class for the events controller pods
+  priorityClassName: ""
+
+  # -- Resource limits and requests for the events controller pods
   resources: {}
-  # -- Event source controller container-level security context
-  containerSecurityContext: {}
-
-eventbusController:
-  # -- Event bus controller name
-  name: eventbus-controller
-  # -- Repository to use for the event bus controller
-  image: argoproj/argo-events
-  # -- Overrides the image tag
-  # @default -- `""` (default is the chart appVersion)
-  tag: ""
-  # -- The number of event bus controller pods to run
-  replicaCount: 1
-  # -- Additional environment variables to pass to event bus controller
-  extraEnv: []
-    # - name: DEBUG_LOG
-    #   value: "true"
-
-  # -- Annotations to be added to event bus controller pods
+  #  limits:
+  #    cpu: 500m
+  #    memory: 512Mi
+  #  requests:
+  #    cpu: 250m
+  #    memory: 256Mi
+
+  # -- Additional containers to be added to the events controller pods
+  extraContainers: []
+
+  # -- Init containers to add to the events controller pods
+  initContainers: []
+
+  serviceAccount:
+    # -- Create a service account for the events controller
+    create: true
+    # -- Service account name
+    name: ''
+    # -- Annotations applied to created service account
+    annotations: {}
+    # -- Automount API credentials for the Service Account
+    automountServiceAccountToken: true
+
+  ## Events controller metrics configuration
+  metrics:
+    # -- Deploy metrics service
+    enabled: false
+    service:
+      # -- Metrics service annotations
+      annotations: {}
+      # -- Metrics service labels
+      labels: {}
+      # -- Metrics service port
+      servicePort: 8082
+    serviceMonitor:
+      # -- Enable a prometheus ServiceMonitor
+      enabled: false
+      # -- Prometheus ServiceMonitor interval
+      interval: 30s
+      # -- Prometheus [RelabelConfigs] to apply to samples before scraping
+      relabelings: []
+      # -- Prometheus [MetricRelabelConfigs] to apply to samples before ingestion
+      metricRelabelings: []
+      # -- Prometheus ServiceMonitor selector
+      selector: {}
+        # prometheus: kube-prometheus
+      # -- Prometheus ServiceMonitor namespace
+      namespace: "" # "monitoring"
+      # -- Prometheus ServiceMonitor labels
+      additionalLabels: {}
+
+## Argo Events admission webhook
+webhook:
+  # -- Enable admission webhook. Applies only for cluster-wide installation
+  enabled: true
```

Comment:
Maybe we should disable it by default as it's documented as an (ALPHA) extension:
https://argoproj.github.io/argo-events/validating-admission-webhook/

Reply:
Sure we can keep it opt-in feature.